buildpack/telemetry/fluentbit.py

```diff
@@ -0,0 +1,131 @@
+import logging
+import os
+import subprocess
+import shutil
+import socket
+
+import backoff
+
+from buildpack import util
+
+
+NAMESPACE = "fluentbit"
+CONF_FILENAME = f"{NAMESPACE}.conf"
+FILTER_FILENAME = "redaction.lua"
+FLUENTBIT_ENV_VARS = {
+    "SPLUNK_PORT": os.getenv("SPLUNK_PORT", default="8088"),
+    "FLUENTBIT_LOGS_PORT": os.getenv("FLUENTBIT_LOGS_PORT", default="5170"),
+}
+
+REQUIRED_FLUENTBIT_ENV_VARS = ["SPLUNK_HOST", "SPLUNK_TOKEN"]
+
+
+def _set_default_env(m2ee):
+    for var_name, value in FLUENTBIT_ENV_VARS.items():
+        util.upsert_custom_environment_variable(m2ee, var_name, value)
+
+
+def stage(buildpack_dir, destination_path, cache_path):
+
+    if not is_splunk_enabled():
+        return
+
+    logging.info("Fluent Bit is being installed...")
+
+    util.resolve_dependency(
+        "fluentbit",
+        # destination_path - DOT_LOCAL_LOCATION
+        os.path.join(destination_path, NAMESPACE),
+        buildpack_dir=buildpack_dir,
+        cache_dir=cache_path,
+    )
+
+    for filename in (CONF_FILENAME, FILTER_FILENAME):
+        shutil.copy(
+            os.path.join(buildpack_dir, "etc", NAMESPACE, filename),
+            os.path.join(
+                destination_path,
+                NAMESPACE,
+            ),
+        )
+
+
+def update_config(m2ee):
+
+    if not is_splunk_enabled():
+        return
+
+    _set_default_env(m2ee)
+
+    util.upsert_logging_config(
+        m2ee,
+        {
+            "type": "tcpjsonlines",
+            "name": "FluentbitSubscriber",
+            "autosubscribe": "INFO",
+            "host": "localhost",
+            "port": FLUENTBIT_ENV_VARS["FLUENTBIT_LOGS_PORT"],
+        },
+    )
+
+
+def run():
+
+    if not is_splunk_enabled():
+        return
+
+    fluentbit_dir = os.path.join(
+        os.path.abspath(".local"),
+        NAMESPACE,
+    )
+
+    fluentbit_bin_path = os.path.join(
+        fluentbit_dir,
+        "fluent-bit",
+    )
+
+    fluentbit_config_path = os.path.join(
+        fluentbit_dir,
+        CONF_FILENAME,
+    )
+
+    if not os.path.exists(fluentbit_bin_path):
+        logging.warning(
+            "Fluent Bit is not installed yet. "
+            "Please redeploy your application to complete "
+            "Fluent Bit installation."
+        )
+        return
+
+    logging.info("Starting Fluent Bit...")
```

Comment:
there is no log line that says splunk is enabled/configured. Isn't it confusing for customers? We mention in the docs that we use fluentbit for splunk but it should be made clear in logs as well IMO.

Reply:
In splunk.py module I added a log line about splunk